src/main/java/eu/europa/ted/efx/xpath/XPathScriptGenerator.java

```diff
@@ -83,10 +83,7 @@ public <T extends Expression> T composeFieldValueReference(PathExpression fieldR
     if ((MultilingualStringExpression.class.isAssignableFrom(type)
         || MultilingualStringListExpression.class.isAssignableFrom(type))
         && !XPathContextualizer.hasPredicate(fieldReference, "@languageID")) {
-      PathExpression languageSpecific = XPathContextualizer.addPredicate(fieldReference, "@languageID=$__LANG__");
-      String script = "(for $__LANG__ in $PREFERRED_LANGUAGES return " + languageSpecific.script
-          + "/normalize-space(text()), " + fieldReference.script + "/normalize-space(text()))[1]";
-      return Expression.instantiate(script, type);
+      return Expression.instantiate("efx:preferred-language-text(" + fieldReference.script + ")", type);
```

Comment:
This implies that the efx:preferred-language-text function should do the "normalize-space", otherwise comparisons with string will not be correct (see the test below).
This should be indicated in the "specification" for this function.

Reply:
Ok Your point is valid. The thing here is even more complicated than that:
a) this is a workaround for the fact that there is no way to select a visualisation language in EFX-1.
b) this implementation in the ScriptGenerator is actually problematic as a workaround because a function in the script generator (which is supposed to be used in both EFX Epression and Template parsing) relies on the affailability of an XSL function (which can only be available if a template is being visualized).

I have transferred this idea from EFX-2 and applied it in an implicit manner, in order to workaround the issue with EFX-1. Bot the more I examine this the more I find problems with any kind of workaround I am using.

I will give this another thought before I merge these changes, but eventually I need to compromise and select a workaround to apply in EFX-1 templates.